chrisj117/CS-472-GROUP-2-PROJECT
# Request School API (GET, POST, DELETE)

**_WHAT DOES THIS PR DO?_**

  - Creates django app ```requestschool```
  - Creates the model and serializer for said app
  - Implements the views GET, POST, DELETE for the API
  - Contains tests for model creation and get, post, delete requests.

**_WHAT ISSUES ARE RELATED TO THIS PR?_**

  - #39 
  - #40
  - #41 
  - #42 

**_HOW DO I TEST OUT THIS PR?_**

Run the project locally, only the backend portion is required to be run.
Endpoint is at: ```http://localhost:8000/api/v1/request-school/```
Django restframework UI makes testing this very straightforward, just open the above link on your browser after running ```python3 manage.py runserver```.
Note: you may need to migrate first for it to successfully work: ```python3 manage.py makemigrations``` -> ```python3 manage.py migrate```

- Get all school requests ```http://localhost:8000/api/v1/request-school/```
- Get a certain school request by school name ```http://localhost:8000/api/v1/request-school/<school_name>```
- Create a school request ```POST http://localhost:8000/api/v1/request-school/```
- - ID field is defaulted, do not fill it out :)
- Delete a school request, DELETE on the endpoint you wish to delete. I.E. create a school request or make sure one exists that has the school name "UNLV", navigate to ```http://localhost:8000/api/v1/request-school/UNLV```, and click the delete red button in the restframework UI.

**_ADDITIONAL INFORMATION_**

Important things to consider from this base school request code:
- Currently there is not PUT request defined. Whether we want a school request to be updated after it has been made or not is up to us.
- In the current state of this code, someone can request for ```UNR``` and then someone else can request for ```University of Nevada, Reno``` and these would be technically separate requests, despite referring to the same school. There are upsides to keeping this functionality, as checking duplicates can be potentially very difficult considering the number of universities that exist.
- The model for ```requestschool``` only contains a ```school_name``` and ```website```. We may want to add a ```school_short_name``` at some point potentially.
- This PR creates a new Django app, but we may want to refactor this app to be included within the ```school``` app later!
